etc/profile-a-l/keepassxc.profile

```diff
@@ -62,6 +62,7 @@ include disable-xdg.inc
 #whitelist ${HOME}/.config/KeePassXCrc
 #include whitelist-common.inc
 
+whitelist ${RUNUSER}/wayland-*
 whitelist /usr/share/keepassxc
 include whitelist-run-common.inc
```

Comment:
```suggestion
whitelist /usr/share/keepassxc
include whitelist-run-common.inc
include whitelist-runuser-common.inc
```

Does this fix the issue?

Reply:
Yes it does. Thanks!

Reply:
Is there some way of putting this in a local config file? If I edit `/etc/firejail/keepassxc.profile` directly, the change may be overwritten on next update...

Reply:
https://github.com/netblue30/firejail/wiki/Creating-overrides

Reply:
> > Does this fix the issue?
>
> Yes it does. Thanks!

It still does not make sense because `keepassxc.profile` does not have any whitelisting for RUNUSER and should therefore default to blacklisting only. This can fix it only through a side-effect and if this is the case we should fix the real cause.

Also DO NOT MERGE any PR enabling whitelisting in RUNUSER for kpxc without careful consideration.

Reply:
@rusty-snake [on Dec 30](https://github.com/netblue30/firejail/pull/5454#discussion_r1059512589):

> > > Does this fix the issue?
> > 
> > Yes it does. Thanks!
> 
> It still does not make sense because `keepassxc.profile` does not have any
> whitelisting for RUNUSER and should therefore default to blacklisting only.
> This can fix it only through a side-effect and if this is the case we should
> fix the real cause.

Yeah, looking at it again, this issue shouldn't happen with the default
profiles.

From my testing with firejail-git, I cannot reproduce this; all paths in the
root of `${RUNUSER}` are present (even if some are blacklisted by default) with
the default keepassxc.profile:

```console
$ find "/run/user/$(id -u)" -maxdepth 1 | wc -l
10
$ firejail --quiet --ignore=private-bin \
  --ignore='include keepassxc.local' --ignore='include globals.local' \
  --profile=keepassxc \
  find "/run/user/$(id -u)" -maxdepth 1 | wc -l
10
```

Maybe firejail 0.9.70 doesn't automatically mount `${RUNUSER}` when
whitelisting /run.

Anyway, I posted the above suggestion because whitelist-runuser-common.inc
contains the following line:

```
whitelist ${RUNUSER}/wayland-?
```

Which is similar to the line that the PR adds:

```
whitelist ${RUNUSER}/wayland-*
```

> Also DO NOT MERGE any PR enabling whitelisting in RUNUSER for kpxc without
> careful consideration.

I'm personally in no hurry to merge this, especially before the root cause is
found.

From my testing, I see that including whitelist-runuser-common.inc makes less
paths visible in `${RUNUSER}`.

To clarify, are you thinking of any specific program/integration issue that
this could cause?

Reply:
> To clarify, are you thinking of any specific program/integration issue that
this could cause?

- Browser Integration
- ssh-agent
- ...

KPXC has support to provided secrets via a lot of ways. The most use sockets in RUNUSER. Some sockets depend on the distro or other conditions. While not all of them work out-of-the-box with the current profile, we should not make the situation worse than it is.